[bitnami/mariadb-galera] node failure: donor hits "Access denied for mariabackup"

**Which chart**:
name = `bitnami/mariadb-galera`
version = `6.0.1`

**Describe the bug**
After a node failure, the cluster cannot recover because WSREP syncing fails.

**To Reproduce**
Steps to reproduce the behavior:

1. [Install](https://bitnami.com/stack/mariadb-galera/helm) the chart with default values, except `replicaCount`: e.g. `helm install galera bitnami/mariadb-galera --set replicaCount=2`
2. Insert some data into the cluster (add databases, etc);
3. Wait for some time (eg. 1 week): this seems not reproducible on freshly installed clusters, but I did manage to replicate on two independent production clusters (totally different kubernetes clusters);
3. Simulate a node failure by scaling down the `StatefulSet` and removing the volume: 
```
kubectl scale sts galera-mariadb-galera --replicas=1
kubectl delete pvc data-galera-mariadb-galera-1 # this removes the PV, too (checked)
kubectl scale sts galera-mariadb-galera --replicas=2
```
4. The failed replica enters a crash loop, which can be checked with `kubectl get pod`
```
NAME                      READY   STATUS             RESTARTS   AGE
galera-mariadb-galera-0   1/1     Running            0          15m32s
galera-mariadb-galera-1   0/1     CrashLoopBackOff   4          3m36s
```
(see below for further investigation results)

**Expected behavior**
Expected the failed node to re-sync with the cluster automatically.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.7.0", GitCommit:"eeac83883cb4014fe60267ec6373570374ce770b", GitTreeState:"clean", GoVersion:"go1.16.8"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-18T02:34:11Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

I managed to collect the logs from two independent production clusters. In the **donor** we see the `Access denied` entry
```
WSREP_SST: [INFO] Streaming with mbstream (20211128 17:57:09.855)
WSREP_SST: [INFO] Using socat as streamer (20211128 17:57:09.860)
WSREP_SST: [INFO] Using '/opt/bitnami/mariadb/tmp/tmp.Bh5w0LMuYV' as mariabackup temporary directory (20211128 17:57:09.872)
WSREP_SST: [INFO] Using '/tmp/tmp.1Y8o8s6jCx' as mariabackup working directory (20211128 17:57:09.877)
WSREP_SST: [INFO] Streaming GTID file before SST (20211128 17:57:09.882)
WSREP_SST: [INFO] Evaluating '/opt/bitnami/mariadb//bin/mbstream' -c 'xtrabackup_galera_info' | socat -u stdio TCP:10.0.1.190:4444; RC=( ${PIPESTATUS[@]} ) (20211128 17:57:09.887)
WSREP_SST: [INFO] Sleeping before data transfer for SST (20211128 17:57:09.894)
2021-11-28 17:57:11 0 [Note] WSREP: (58c80db0-8ccd, 'tcp://0.0.0.0:4567') turning message relay requesting off
WSREP_SST: [INFO] Streaming the backup to joiner at 10.0.1.190:4444 (20211128 17:57:19.899)
WSREP_SST: [INFO] Evaluating /opt/bitnami/mariadb//bin/mariabackup --defaults-file='/opt/bitnami/mariadb/conf/my.cnf'   --backup --no-version-check --databases-exclude='lost+found'  --tmpdir='/opt/bitnami/mariadb/tmp/tmp.Bh5w0LMuYV'  --innodb-data-home-dir='./' --log-bin='mysql-bin' --user='mariabackup' --socket='/opt/bitnami/mariadb/tmp/mysql.sock' --galera-info --stream=mbstream --target-dir='/tmp/tmp.1Y8o8s6jCx' --datadir='/bitnami/mariadb/data' --mysqld-args '--basedir=/opt/bitnami/mariadb' '--pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid' '--wsrep-node-name=galera-mariadb-galera-1' '--wsrep-node-address=10.0.1.142' '--wsrep-cluster-name=galera' '--wsrep-cluster-address=gcomm://' '--wsrep-sst-method=mariabackup' '--wsrep-sst-auth=mariabackup:<hidden>' 2> '/bitnami/mariadb/data/mariabackup.backup.log' | socat -u stdio TCP:10.0.1.190:4444; RC=( ${PIPESTATUS[@]} ) (20211128 17:57:19.902)
2021-11-28 17:57:19 20 [Warning] Access denied for user 'mariabackup'@'localhost' (using password: YES)
WSREP_SST: [ERROR] mariabackup finished with error: 1.  Check syslog or '/bitnami/mariadb/data/mariabackup.backup.log' for details (20211128 17:57:19.917)
WSREP_SST: [ERROR] Cleanup after exit with status:22 (20211128 17:57:19.921)
```
and in the **joiner** the process fails:
```
2021-11-28 17:33:00 0 [Warning] WSREP: 1.0 (galera-mariadb-galera-1): State transfer to 0.0 (galera-mariadb-galera-2) failed: -22 (Invalid argument)
2021-11-28 17:33:00 0 [ERROR] WSREP: /bitnami/blacksmith-sandox/libgalera-26.4.9/gcs/src/gcs_group.cpp:gcs_group_handle_join_msg():1170: Will never receive state. Need to abort.
2021-11-28 17:33:00 0 [Note] WSREP: gcomm: terminating thread
2021-11-28 17:33:00 0 [Note] WSREP: gcomm: joining thread
2021-11-28 17:33:00 0 [Note] WSREP: gcomm: closing backend
2021-11-28 17:33:00 0 [ERROR] WSREP: Process completed with error: wsrep_sst_mariabackup --role 'joiner' --address '10.0.0.41' --datadir '/bitnami/mariadb/data/' --defaults-file '/opt/bitnami/mariadb/conf/my.cnf' --parent '1' --binlog 'mysql-bin' --mysqld-args --defaults-file=/opt/bitnami/mariadb/conf/my.cnf --basedir=/opt/bitnami/mariadb --datadir=/bitnami/mariadb/data --socket=/opt/bitnami/mariadb/tmp/mysql.sock --pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid --wsrep-node-name=galera-mariadb-galera-2 --wsrep-node-address=10.0.0.41 --wsrep-cluster-name=galera --wsrep-cluster-address=gcomm://galera-mariadb-galera-headless.mariadb-galera.svc.cluster.local --wsrep-sst-method=mariabackup --wsrep-sst-auth=mariabackup:<hidden> --old-mode=: 2 (No such file or directory)
2021-11-28 17:33:00 0 [ERROR] WSREP: Failed to read uuid:seqno and wsrep_gtid_domain_id from joiner script.
```

Full logs attached: [donor](https://github.com/bitnami/charts/files/7614234/scratch-kPslbi.txt) and [joiner](https://github.com/bitnami/charts/files/7614230/scratch-gwxmYg.txt).

This is happening again.

The cluster in question is a 3-node galera cluster, setup as per OP. Node 0 is crash-looping with an error very similar (if not identical) to the one described in this issue. [Attached](https://github.com/bitnami/charts/files/7706952/scratch-HlOsvv.txt) is the complete crash log.

From which I would like to highlight the following snippet:
```
[...]
2021-12-13 20:32:42 0 [Note] WSREP: Member 2.0 (galera-mariadb-galera-0) requested state transfer from '*any*'. Selected 0.0 (galera-mariadb-galera-1)(SYNCED) as donor.
2021-12-13 20:32:42 0 [Note] WSREP: Shifting PRIMARY -> JOINER (TO: 65715)
2021-12-13 20:32:42 2 [Note] WSREP: Requesting state transfer: success, donor: 0
2021-12-13 20:32:42 2 [Note] WSREP: Resetting GCache seqno map due to different histories.
[...]
2021-12-13 20:32:52 0 [ERROR] WSREP: Process completed with error: wsrep_sst_mariabackup --role 'joiner' --address '10.0.0.26' --datadir '/bitnami/mariadb/data/' --defaults-file '/opt/bitnami/mariadb/conf/my.cnf' --parent '1' --binlog 'mysql-bin' --mysqld-args --defaults-file=/opt/bitnami/mariadb/conf/my.cnf --basedir=/opt/bitnami/mariadb --datadir=/bitnami/mariadb/data --socket=/opt/bitnami/mariadb/tmp/mysql.sock --pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid --wsrep-node-name=galera-mariadb-galera-0 --wsrep-node-address=10.0.0.26 --wsrep-cluster-name=galera --wsrep-cluster-address=gcomm://galera-mariadb-galera-headless.mariadb-galera.svc.cluster.local --wsrep-sst-method=mariabackup --wsrep-sst-auth=mariabackup:<redacted> --old-mode=: 2 (No such file or directory)
2021-12-13 20:32:52 0 [ERROR] WSREP: Failed to read uuid:seqno and wsrep_gtid_domain_id from joiner script.
[...]
2021-12-13 20:32:52 2 [ERROR] WSREP: Application state transfer failed. This is unrecoverable condition, restart required.
```

So apparently this time around the problem is `(No such file or directory)`. However, I cannot seem to find further information on what file (or directory) may be missing.

Since I do not see any permission problems this time, this seems to contradict [my hypothesis](https://github.com/bitnami/charts/issues/8256#issuecomment-989715754). Also, I can confirm that the password used for `--wsrep-sst-auth` is the same I manually specified in the chart.

Can someone please suggest some debugging steps to figure out what the problem can be? I am out of ideas.

Sorry, I wrote too soon. Inspecting the donor (see [attached log](https://github.com/bitnami/charts/files/7707034/scratch-dNkxsy.txt)) reveals the same `Access denied` problem, and also an `Invalid argument`:

```
[...]
WSREP_SST: [INFO] Streaming the backup to joiner at 10.0.0.26:4444 (20211213 21:04:44.384)
WSREP_SST: [INFO] Evaluating /opt/bitnami/mariadb//bin/mariabackup --defaults-file='/opt/bitnami/mariadb/conf/my.cnf'   --backup --no-version-check --databases-exclude='lost+found'  --tmpdir='/opt/bitnami/mariadb/tmp/tmp.e4BHW4zRNt'  --innodb-data-home-dir='./' --log-bin='mysql-bin' --user='mariabackup' --socket='/opt/bitnami/mariadb/tmp/mysql.sock' --galera-info --stream=mbstream --target-dir='/tmp/tmp.LH3EUUp3Yu' --datadir='/bitnami/mariadb/data' --mysqld-args '--basedir=/opt/bitnami/mariadb' '--pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid' '--wsrep-node-name=galera-mariadb-galera-1' '--wsrep-node-address=10.0.1.88' '--wsrep-cluster-name=galera' '--wsrep-cluster-address=gcomm://galera-mariadb-galera-headless.mariadb-galera.svc.cluster.local' '--wsrep-sst-method=mariabackup' '--wsrep-sst-auth=mariabackup:<redacted>' '--old-mode=' 2> '/bitnami/mariadb/data/mariabackup.backup.log' | socat -u stdio TCP:10.0.0.26:4444; RC=( ${PIPESTATUS[@]} ) (20211213 21:04:44.387)
2021-12-13 21:04:44 184156 [Warning] Access denied for user 'mariabackup'@'localhost' (using password: YES)
WSREP_SST: [ERROR] mariabackup finished with error: 1.  Check syslog or '/bitnami/mariadb/data/mariabackup.backup.log' for details (20211213 21:04:44.410)
WSREP_SST: [ERROR] Cleanup after exit with status:22 (20211213 21:04:44.413)
[...]
2021-12-13 21:04:44 0 [Note] WSREP: Server status change donor -> joined
2021-12-13 21:04:44 0 [Note] WSREP: wsrep_notify_cmd is not defined, skipping notification.
2021-12-13 21:04:44 0 [ERROR] WSREP: Command did not run: wsrep_sst_mariabackup --role 'donor' --address '10.0.0.26:4444/xtrabackup_sst//1' --local-port '3306' --socket '/opt/bitnami/mariadb/tmp/mysql.sock' --datadir '/bitnami/mariadb/data/' --defaults-file '/opt/bitnami/mariadb/conf/my.cnf' --gtid '5397831d-507e-11ec-b9bb-e70e5db89b6b:65890' --gtid-domain-id '0' --binlog 'mysql-bin' --mysqld-args --defaults-file=/opt/bitnami/mariadb/conf/my.cnf --basedir=/opt/bitnami/mariadb --datadir=/bitnami/mariadb/data --socket=/opt/bitnami/mariadb/tmp/mysql.sock --pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid --wsrep-node-name=galera-mariadb-galera-1 --wsrep-node-address=10.0.1.88 --wsrep-cluster-name=galera --wsrep-cluster-address=gcomm://galera-mariadb-galera-headless.mariadb-galera.svc.cluster.local --wsrep-sst-method=mariabackup --wsrep-sst-auth=mariabackup:<redacted> --old-mode=
2021-12-13 21:04:44 0 [Note] WSREP: Donor monitor thread ended with total time 10 sec
2021-12-13 21:04:44 0 [Warning] WSREP: 0.0 (galera-mariadb-galera-1): State transfer to 1.0 (galera-mariadb-galera-0) failed: -22 (Invalid argument)
2021-12-13 21:04:44 0 [Note] WSREP: Shifting DONOR/DESYNCED -> JOINED (TO: 65890)
[...]
```

Again though, the password printed in the logs is the correct one. How can I recover from this situation? Is there a way to manually test whatever the sync process is doing (eg. trying authentication or backup) and to get to the bottom of this?

--- a/bitnami/mariadb-galera/Chart.yaml
+++ b/bitnami/mariadb-galera/Chart.yaml
@@ -27,4 +27,4 @@ sources:
   - https://github.com/bitnami/bitnami-docker-mariadb-galera
   - https://github.com/prometheus/mysqld_exporter
   - https://mariadb.org
-version: 6.0.10
+version: 6.0.11

--- a/bitnami/mariadb-galera/templates/NOTES.txt
+++ b/bitnami/mariadb-galera/templates/NOTES.txt
@@ -85,3 +85,17 @@ To upgrade this helm chart:
 
 {{ include "mariadb-galera.validateValues" . }}
 {{ include "mariadb-galera.checkRollingTags" . }}
+{{- $requiredPasswords := list -}}
+{{- $secretName := include "mariadb-galera.secretName" . -}}
+{{- if not .Values.existingSecret -}}
+  {{- $requiredRootPassword := dict "valueKey" "rootUser.password" "secret" $secretName "field" "mariadb-root-password" -}}
+  {{- $requiredPasswords = append $requiredPasswords $requiredRootPassword -}}
+  {{- $requiredMariaBackupPassword := dict "valueKey" "galera.mariabackup.password" "secret" $secretName "field" "mariadb-galera-mariabackup-password" -}}
+  {{- $requiredPasswords = append $requiredPasswords $requiredMariaBackupPassword -}}
+  {{- if .Values.db.user }}
+    {{- $requiredUserPassword := dict "valueKey" "db.password" "secret" $secretName "field" "mariadb-password" -}}
+    {{- $requiredPasswords = append $requiredPasswords $requiredUserPassword -}}
+  {{- end -}}
+{{- end -}}
+{{- $requiredRedisPasswordErrors := include "common.validations.values.multiple.empty" (dict "required" $requiredPasswords "context" $) -}}
+{{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" (list $requiredRedisPasswordErrors) "context" $) -}}

--- a/bitnami/mariadb-galera/values.yaml
+++ b/bitnami/mariadb-galera/values.yaml
@@ -64,7 +64,7 @@ diagnosticMode:
 image:
   registry: docker.io
   repository: bitnami/mariadb-galera
-  tag: 10.6.5-debian-10-r29
+  tag: 10.6.5-debian-10-r35
   ## Specify a imagePullPolicy
   ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
   ## ref: https://kubernetes.io/docs/user-guide/images/#pre-pulling-images
@@ -699,7 +699,7 @@ metrics:
   image:
     registry: docker.io
     repository: bitnami/mysqld-exporter
-    tag: 0.13.0-debian-10-r185
+    tag: 0.13.0-debian-10-r191
     pullPolicy: IfNotPresent
     ## Optionally specify an array of imagePullSecrets (secrets must be manually created in the namespace)
     ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

